skyplane/broadcast/bc_client.py

```diff
@@ -0,0 +1,117 @@
+import uuid
+from datetime import datetime
+from pathlib import Path
+
+from typing import TYPE_CHECKING, Optional, List
+
+from skyplane.api.client import tmp_log_dir
+from skyplane.api.usage.client import get_clientid
+from skyplane.api.dataplane import Dataplane
+from skyplane.broadcast import BroadcastDataplane
+from skyplane.broadcast.bc_planner import BroadcastDirectPlanner, BroadcastMDSTPlanner, BroadcastHSTPlanner, BroadcastILPSolverPlanner
+from skyplane.api.impl.provisioner import Provisioner
+from skyplane.api.transfer_config import TransferConfig
+from skyplane.utils import logger
+
+if TYPE_CHECKING:
+    from skyplane.api.auth_config import AWSConfig, AzureConfig, GCPConfig
+
+
+class SkyplaneBroadcastClient:
+    def __init__(
+        self,
+        aws_config: Optional["AWSConfig"] = None,
+        azure_config: Optional["AzureConfig"] = None,
+        gcp_config: Optional["GCPConfig"] = None,
+        transfer_config: Optional[TransferConfig] = None,
+        log_dir: Optional[str] = None,
+    ):
+        self.clientid = get_clientid()
+        self.aws_auth = aws_config.make_auth_provider() if aws_config else None
+        self.azure_auth = azure_config.make_auth_provider() if azure_config else None
+        self.gcp_auth = gcp_config.make_auth_provider() if gcp_config else None
+        self.transfer_config = transfer_config if transfer_config else TransferConfig()
+        self.log_dir = (
+            tmp_log_dir / "transfer_logs" / f"{datetime.now().strftime('%Y%m%d_%H%M%S')}-{uuid.uuid4().hex[:8]}"
```

Comment:
consider adding Z to the end of the format string to indicate the timestamp as UTC:

> If the time is in [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time), add a Z directly after the time without a space. Z is the zone designator for the zero UTC offset. "09:30 UTC" is therefore represented as "09:30Z" or "T0930Z". "14:45:15 UTC" would be "14:45:15Z" or "T144515Z".
>
> The Z suffix in the ISO 8601 time representation is sometimes referred to as "Zulu time" or "Zulu meridian" because the same letter is used to designate the [Zulu time zone](https://en.wikipedia.org/wiki/List_of_military_time_zones).
from [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Coordinated_Universal_Time_(UTC)) on wikipedia